init/cmdline.go

```diff
@@ -127,23 +127,29 @@ func parseParams(params string) error {
 				return fmt.Errorf("invalid UUID %s %v", parts[0], err)
 			}
 
-			dev := luksMapping{
-				ref:  &deviceRef{refFsUUID, uuid},
-				name: parts[1],
-			}
-			luksMappings = append(luksMappings, dev)
+			var o = findOrCreateLuksMapping(uuid)
+			o.name = parts[1]
 		case "rd.luks.uuid":
 			stripped := stripQuotes(value)
-			u, err := parseUUID(stripped)
+			uuid, err := parseUUID(stripped)
```

Comment:
this part needs to be rebased on top of `master` that contains your other changes